### PR TITLE
Add matrix-free laplace code and test

### PR DESCRIPTION
**Pull request by Rombur**
This is the matrix-free equivalent of the `laplace` code

### CODE REVIEW

File: tests/test_laplace_matrix_free.cc [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  Source<dim> source;
+
+  Laplace<dim, fe_degree, double> laplace(MPI_COMM_WORLD);
+  laplace.setup_system(boost::property_tree::ptree(), material_property);
+  laplace.assemble_rhs(source);
+
+  dealii::PreconditionIdentity preconditioner;
+  laplace.solve(preconditioner);
+
+  // The exact solution is quadratique so the error should be zero.
```
Comment (aprokop) [inline tests/test_laplace_matrix_free.cc]: No French! :)

File: tests/test_laplace_matrix_free.cc [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
```
Comment (aprokop) [inline tests/test_laplace_matrix_free.cc]: French

File: tests/test_laplace_matrix_free.cc (hunk 1 above)
Comment (Rombur) [reply tests/test_laplace_matrix_free.cc]: That's the fancy spelling, it adds a  je ne sais quoi  to the library.

**Comment by Rombur** [comment]
>  1.   It seems that one cannot include both laplace.hpp and laplace_matrix_free.hpp due to clashing Laplace class names.

Sure, I will change the names

>  2.   There seems to be a lot of commonalities between two implementations of Laplace. Do you think it would be hard to merge them together with a matrix-free being a parameter option?

That's what I thought at first too but when you compare the code more closely there is actually little overlap except to create the geometry

**Comment by aprokop** [comment]
:( Please do not rebase PRs until approved.

**Comment by Rombur** [comment]
But you approved the PR! I was going to merge myself...

**Comment by aprokop** [comment]
Lol, I forgot about it, and wanted to check the changes. Mea culpa.

**Comment by aprokop** [comment]
Why did you need to update the tolerance?

**Comment by Rombur** [comment]
The tester was not working since we updated the docker images. The build was crashing during the configuration but jenkins reported that everything was fine....

**Pull request merged**